Source/WebCore/platform/MediaStrategy.cpp

```diff
@@ -83,4 +87,30 @@ void MediaStrategy::addMockMediaSourceEngine()
 }
 #endif
 
+#if ENABLE(VIDEO)
+RefPtr<AudioVideoRenderer> MediaStrategy::createAudioVideoRenderer(WTF::LoggerHelper* loggerHelper) const
+{
+#if USE(AVFOUNDATION)
+    ASSERT(loggerHelper);
+    return AudioVideoRendererAVFObjC::create(Ref { loggerHelper->logger() }, loggerHelper->logIdentifier());
+#else
+    UNUSED_PARAM(loggerHelper);
+    return nullptr;
+#endif
+}
+
+bool MediaStrategy::hasRemoteRendererFor(MediaPlayerMediaEngineIdentifier type) const
+{
+    return m_remoteRenderersEnabled & 1 << static_cast<uint32_t>(type);
```

Comment:
Can `m_remoteRenderersEnabled` be an `OptionSet< MediaPlayerMediaEngineIdentifier >` instead of a `uint32_t`  so you don't need to `static_cast`?

Reply:
MediaPlayerMediaEngineIdentifier is currently an enum, if I was to change it for one usable in an OptionSet it would no longer work when sending over IPC. As it must be a plain enum or an OptionSet; it can't be both with the current serialiser.